pkg/template/engine.go

```diff
@@ -30,8 +32,5 @@ func EngineForVersion(version esapi.TemplateEngineVersion) (ExecFunc, error) {
 	case esapi.TemplateEngineV2:
 		return v2.Execute, nil
 	}
-
-	// in case we run with a old v1alpha1 CRD
-	// we must return v1 as default
-	return v1.Execute, nil
+	return nil, fmt.Errorf("unsupported template engine version: %s", version)
```

Comment:
This change might not be correct @Skarlso, because I guess its possible that if someone upgrades from `v1alpha1` to `v1beta1` it might not correctly set the default.

I might need to rever this part.

Reply:
I made a PR to fix this, while still keeping the check for actually invalid values: https://github.com/external-secrets/external-secrets/pull/4182